### ISSUE TITLE
sql/schemachanger: remove remaining limitations for declarative schema changer

### ISSUE THREAD

**Issue by fqazi** [cockroachdb/cockroach#80545]
The declarative schema changer is currently disabled when adding columns in the following scenarios:

- [ ] Legacy schema changer will be used whenever serial and generated columns are observed.
- [ ] Regional by row tables will default to the legacy schema changer since we are missing proper support for updating zone configs
- [ ] Columns with the unique constraint will default to the legacy schema changer 
- [ ] Default expressions using sequences will default to the legacy schema changer
- [ ] Better errors can be surfaced on backfill errors for the declarative schema changer, at least one case falls to the legacy one today for this purpose
- [ ] Tests which need the MVCC backfiller are currently using the legacy schema changer

**Comment by postamar** [comment]
Thanks for filing this and collecting all the known odds and ends. Do we also need to fall back to the legacy schema changer for ON UPDATE in addition to DEFAULT?

**Comment by fqazi** [comment]
Yes that's true for all expression types let me fix the TODO @postamar

**Comment by postamar** [comment]
I've unassigned myself considering that I'm very unlikely to close this before going on leave. I think I assigned myself this one in the first place because I was interested in owning the overall story. Perhaps the list in the issue description needs some grooming, I haven't checked.
